pkg/controllers/deprovisioning/consolidation.go

```diff
@@ -241,26 +241,36 @@ func (c *Consolidation) String() string {
 }
 
 func canBeTerminated(node CandidateNode, pdbs *PDBLimits) bool {
-	return node.DeletionTimestamp.IsZero() && pdbs.CanEvictPods(node.pods) && !podsPreventEviction(node)
+	if !node.DeletionTimestamp.IsZero() {
+		return false
+	}
+	if _, ok := pdbs.CanEvictPods(node.pods); !ok {
+		return false
+	}
+
+	if _, ok := PodsPreventEviction(node.pods); ok {
+		return false
+	}
+	return true
 }
 
-// podsPreventEviction returns true if there are pods that would prevent eviction
-func podsPreventEviction(node CandidateNode) bool {
-	for _, p := range node.pods {
+// PodsPreventEviction returns true if there are pods that would prevent eviction
+func PodsPreventEviction(pods []*v1.Pod) (string, bool) {
```

Comment:
optional: a nillable error is a string/bool idiom.
```suggestion
func PodsPreventEviction(pods []*v1.Pod) error {
```